joomla/joomla-cms
# [com_banners] tracks view: Add max level filter

Pull Request for Improvement.
#### Summary of Changes

All backend views that are or are associated with tree like structures (e.g. menus, categories) should have a max level filter.

This is valid currently for:
- com_menus items view
- com_content articles view
- com_content featured view
- com_tags tags view

So, this PR is a part of series of PR to add that filter in the views that use categories:
- com_contacts contacts view (RTC https://github.com/joomla/joomla-cms/pull/9975)
- com_banners banners view (RTC https://github.com/joomla/joomla-cms/pull/9977)
- com_newsfeeds newsfeeds view (RTC https://github.com/joomla/joomla-cms/pull/9978)
- com_notes notes view (RTC https://github.com/joomla/joomla-cms/pull/9984)
- com_banners tracks view (**this PR**)

![image](https://cloud.githubusercontent.com/assets/9630530/14681342/94532492-0718-11e6-962f-a33881d609c7.png)
#### Testing Instructions
1. Use latest staging and apply this patch
2. Create some banners categories with at least two levels
3. Create some banners and associate them to the categories created in 2..
4. Go to "Components" -> "Banners" -> "Tracks" (ensure you have some tracks, see note below) and test the new max levels filter.

Note: If you don't have tracks records you can generate some in the database (`#__banners_tracks` table), example:

``` sql
INSERT INTO `<prefix>_banner_tracks` (`track_date`, `track_type`, `banner_id`, `count`) VALUES ('2016-04-11 00:00:00', '1', '1', '5');
INSERT INTO `<prefix>_banner_tracks` (`track_date`, `track_type`, `banner_id`, `count`) VALUES ('2016-01-11 00:00:00', '2', '1', '7');
INSERT INTO `<prefix>_banner_tracks` (`track_date`, `track_type`, `banner_id`, `count`) VALUES ('2015-04-11 00:00:00', '2', '2', '9');
INSERT INTO `<prefix>_banner_tracks` (`track_date`, `track_type`, `banner_id`, `count`) VALUES ('2016-02-11 00:00:00', '2', '2', '12');
```
